app/helpers/template_helpers.py

```diff
@@ -14,215 +14,299 @@
         )
 
-    default_context = {
-        "lang": language_code,
-        "crest": True,
-        "newTabWarning": lazy_gettext("The following links open in a new tab"),
-        "copyrightDeclaration": {
-            "copyright": lazy_gettext(
-                "Crown copyright and database rights 2021 NIMA MOU577.501."
-            )
-            if theme == "census-nisra"
-            else lazy_gettext("Crown copyright and database rights 2020 OS 100019153."),
-            "text": lazy_gettext(
-                "Use of address data is subject to the terms and conditions."
+    @property
+    def context(self) -> Dict[str, Any]:
+        return {
+            "account_service_url": self._account_service_url,
+            "account_service_log_out_url": self._account_service_log_out_url,
+            "contact_us_url": asdict(
+                Link("Contact us", f"{self._base_url}contact-us/")
             ),
-        },
-        "rows": [
-            {"itemsList": items_list},
-        ],
-        "legal": [
-            {
-                "itemsList": [
-                    {
-                        "text": lazy_gettext("Cookies"),
-                        "url": static_content_urls["cookies"],
-                        "target": "_blank",
-                    },
-                    {
-                        "text": lazy_gettext("Accessibility statement"),
-                        "url": static_content_urls["accessibility_statement"],
-                        "target": "_blank",
-                    },
-                    {
-                        "text": lazy_gettext("Privacy and data protection"),
-                        "url": static_content_urls["privacy_and_data_protection"],
-                        "target": "_blank",
-                    },
-                    {
-                        "text": lazy_gettext("Terms and conditions"),
-                        "url": static_content_urls["terms_and_conditions"],
-                        "target": "_blank",
-                    },
-                ],
-            },
-        ],
-    }
+            "cookie_settings_url": self._cookie_settings_url,
+            "page_header": self.page_header_context,
+            "footer": self.footer_context,
+            "languages": get_languages_context(),
+            "theme": self._map_theme(self._theme),
+            "language_code": get_locale().language,
+            "schema_theme": self._theme,
+            "survey_title": lazy_gettext("Census 2021"),
+            "cdn_url": self._cdn_url,
+            "address_lookup_api_url": self._address_lookup_api,
+            "data_layer": get_data_layer(self._theme),
+            "include_csrf_token": (
+                request.url_rule and "POST" in request.url_rule.methods
+            ),
+            "google_tag_manager_id": self._google_tag_manager_id,
+            "google_tag_manager_auth": self._google_tag_manager_auth,
+        }
+
+    @cached_property
```

Comment:
Is this method ever called more than once per request?

Reply:
No - I'll remove the cache

Reply:
Done